_posts/en/newsletters/2020-08-05-newsletter.md

```diff
@@ -0,0 +1,296 @@
+  order for a UTXO to be spent.  The new language is based on the
+  [miniscript][topic miniscript] policy language but adds the ability to
+  use variables and functions along with several other features.  Minsc
+  policies can be compiled to miniscript policies, which can be further transformed
+  into miniscript and regular Bitcoin Script.  Compatibility
+  with miniscript means that policies developed using Minsc will be
+  solvable using any miniscript-aware wallet in the future, allowing
+  wallets to contribute signatures, preimages, or other data needed to
+  spend bitcoins without their developers having to manually specify how
+  to handle each specific script template the wallet supports.  This
+  should allow faster upgrades to new language features and greatly
+  simplify development of interoperable wallets for coinjoins, contract
+  protocols, shared coin ownership, and other types of desirable
+  collaborations.
+
+    Ivgi has also created an outstanding [website][min.sc] for the language
+    that includes a plethora of examples and a live compiler that allows
+    linking to its input so that developers can easily play with the
+    language and share their Minsc policies with other developers.  We
+    recommend anyone interested in developing spending policies visit the
+    website but, as an illustration of what Minsc can do, we offer the
+    following example adapted from Ivgi's own examples.  Several years
+    ago, before miniscript or Minsc, LN developers hand crafted the
+    following [HTLC script][] specified in BOLT3:
+
+    ```text
+    # To remote node with revocation key
+    OP_DUP OP_HASH160 <RIPEMD160(SHA256(revocationpubkey))> OP_EQUAL
+    OP_IF
+        OP_CHECKSIG
+    OP_ELSE
+        <remote_htlcpubkey> OP_SWAP OP_SIZE 32 OP_EQUAL
+        OP_IF
+            # To local node via HTLC-success transaction.
+            OP_HASH160 <RIPEMD160(payment_hash)> OP_EQUALVERIFY
+            2 OP_SWAP <local_htlcpubkey> 2 OP_CHECKMULTISIG
+        OP_ELSE
+            # To remote node after timeout.
+            OP_DROP <cltv_expiry> OP_CHECKLOCKTIMEVERIFY OP_DROP
+            OP_CHECKSIG
+        OP_ENDIF
+    OP_ENDIF
+    ```
+
+    The same encumbrance can be [specified][htlc minsc] using the following Minsc
+    policy (setting the `cltv_expiry` to 3 hours):
+
+    ```rust
+    fn htlc_received($revocationpubkey, $local_htlcpubkey,
+                     $remote_htlcpubkey, $payment_hash,
+                     $cltv_expiry)
+    {
+      // To local node via HTLC-success transaction
+      $success = pk($local_htlcpubkey) && hash160($payment_hash);
+
+      // To remote node after timeout
+      $timeout = older($cltv_expiry);
+
+      // To remote node with revocation key, or use success/timeout
+      pk($revocationpubkey) || (pk($remote_htlcpubkey) && ($success || $timeout))
+    }
+
+    htlc_received(A, B, C, H, 3 hours)
+    ```
+
+    The Minsc policy is not only much easier for most developers to
```

Comment:
The Minsc policy is ~not only~ much easier for most developers to analyze ~but~ *and* it can take